block-explorer/v3.yml

```diff
@@ -471,6 +471,54 @@ paths:
               schema:
                 $ref: '#/components/schemas/NotFound'
 
+  /addresses/{address}/snapshots:
+    get:
+      tags:
+        - Currency snapshot
+      summary: Get currency snapshots by owner address
+      parameters:
```

Comment:
Do we allow any sorting parameters? What's the default sort?

Reply:
In API endpoints that return collections, we do not support any custom sorting. For example `/transactions`:

<img width="1111" alt="image" src="https://github.com/user-attachments/assets/12897f22-0b04-47ff-8272-15767ef909cd">

So we support `limit`, `search_after/search_before` or `next`.

For  `addresses/{address}/snapshots` I decided to go for `limit` and `next` but without `search_after/search_before`. It is like that because we can't use `hash/ordinal` as `search_after/search_before`. The reason is that `hash/ordinal` are unique within a single metagraph but are not unique across all the metagraphs. 

For example, if you use `hash` as `search_before`, you never know from which snapshot it starts to search because you can have `n` snapshots with the same hash globally.

Reply:
About your second question. The default sort is by `ordinal` and then by `identifier`. So:

```
{ id: "A", ordinal: 1 },
{ id: "B", ordinal: 1 },
{ id: "B", ordinal: 1 },
{ id: "A", ordinal: 2 },
{ id: "C", ordinal: 2 },
```

Reply:
Also worth to mention: `search_after` pagination that we use is stateless so the results may change in between your requests if new documents from new metagraphs get indexed on previous pages and may "push" some snapshots to the next pages.

The solution for that is stateful `scroll pagination`: https://opensearch.org/docs/latest/search-plugins/searching-data/paginate/#scroll-search but it is a little bit tricky because you need to tell OpenSearch how long you want to be querying (e.g. 10 minutes) so it creates a session for you. Then even if new documents get indexed in between requests - it keeps the state of your pagination for 10 minutes.